### PR TITLE
chore: add extended request ID to general AWS Attributes

### PR DESCRIPTION
**Pull request by trivikr**
Fixes https://github.com/open-telemetry/semantic-conventions/issues/1670

## Changes

Adds extended request ID to general AWS Attributes under `aws.extended_request_id`

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Comment by trivikr** [comment]
From my understanding, the `schema-next.yaml` is not required to be changed, as no existing attribute is being updated. This PR adds a new attribute.

Docs: https://opentelemetry.io/docs/specs/otel/schemas/file_format_v1.0.0/#span_events-section

File: model/aws/registry.yaml [hunk 1]
```diff
@@ -12,6 +12,12 @@ groups:
         examples:
           - 79b9da39-b7ae-508a-a6bc-864b2829c622
           - C9ER4AJX75574TDJ
+      - id: aws.extended_request_id
```
Comment (lmolkova) [inline model/aws/registry.yaml:15]: is this name used by AWS? If not, suggesting `aws.response.id`

**Review by lmolkova — COMMENTED** [review]
Please reference this attribute from AWS spans you expect the attribute to be available on - there are quite a few of them defined here https://github.com/open-telemetry/semantic-conventions/blob/main/model/aws/sdk-spans.yml

Also, is it populated by any instrumentations?

File: model/aws/registry.yaml (hunk 1 above)
Comment (trivikr) [reply model/aws/registry.yaml:15]: Yes. The developer guide and JS SDK v3 sample code is shared in description of https://github.com/open-telemetry/semantic-conventions/issues/1670

**Comment by trivikr** [comment]
> is it populated by any instrumentations?

I've been maintaining AWS SDK for JavaScript for 5+ years, and was recently added as component owner of it's OpenTelemetry Instrumentation in https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2557. I plan to add the attribute as soon as it's merged and released in semantic convention.

AWS SDK for Go Observability solution does use it https://github.com/aws/aws-sdk-go-v2/blob/a164d844a90d8b9f86f8cb827b3fef512eabaabf/service/internal/s3shared/metadata_retriever.go#L53

**Comment by trivikr** [comment]
> Please reference this attribute from AWS spans you expect the attribute to be available on

Added in [`24ada32` (#1671)](https://github.com/open-telemetry/semantic-conventions/pull/1671/commits/24ada32145a5e9d75cca97ee44ad43dcff02a28c)

**Comment by trivikr** [comment]
The `markdown-link-check` either incorrectly checks the link from the logs, or the link might have been down when the test was performed.

```console
  ERROR: 1 dead links found in ./docs/system/system-metrics.md !
  [✖] https://blogs.oracle.com/linux/post/understanding-linux-kernel-memory-statistics → Status: 0
make: *** [Makefile:72: markdown-link-check] Error 1
```

It's failing for other PRs too:
* https://github.com/open-telemetry/semantic-conventions/actions/runs/12139916442/job/33848539883
* https://github.com/open-telemetry/semantic-conventions/actions/runs/12293003057/job/34324346286

Can you re-run the workflow?

**Review by lmolkova — APPROVED** [review]

File: .chloggen/aws.extended_request_id.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the area of concern in the attributes-registry, (e.g. http, cloud, db)
+component: aws
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add extended request ID to general AWS Attributes
```
Comment (trask) [inline .chloggen/aws.extended_request_id.yaml:13]:
```suggestion
note: Add extended request ID to general AWS attributes as `aws.extended_request_id`
```

**Review by trask — APPROVED** [review]

**Comment by damemi** [comment]
@trivikr hi, could you please clarify if this header should be required for all AWS services? In your [initial issue](https://github.com/open-telemetry/semantic-conventions/issues/1670), you mentioned that `x-amz-id-2` is used in S3, but here it was also added to services like DynamoDB

I just opened [another issue](https://github.com/open-telemetry/semantic-conventions/issues/2631) (before I saw this) because it doesn't seem like Dynamo uses `x-amz-id-2` in any of its documentation or client code. Should this attribute be removed from the required list for AWS services that don't use it?

**Pull request merged**